### DISCUSSION TITLE
Help with Exiv2 0.28.0 for Windows

### DISCUSSION THREAD

**Discussion by dvongrad** [Exiv2/exiv2#2744] [Q&A]
I apologize in advance if I should have asked this elsewhere or if it has already been answered in a different discussion.

I've been using exiv2 since 2005 when I created an MFC dialog app to rename my pics from Exif date info. Throughout the years, my app got updated and the last update was about 3 years ago when I think the current version of Exiv2 at the time was 0.26.x.

I'm now rewriting the app for new features and efficiency. I had hoped to use 0.28.0 but when I build the app, I get several errors starting with "(': illegal token on right side of '::'" on line 1221 of value.hpp. I downgraded to 0.27.7 and, while it successfully compiles, I am facing inconsistent "No such file or directory" or "Invalid argument" exceptions when executing Exiv2::ImageFactory::open from within my app.

While trying to diagnose the issue, I took the exifprint.cpp example code, put it verbatim in a new C++ console app that successfully created an EXE. Running it with a full path to a JPG file argument, I got the "No such file or directory" error the first time. Trying it with two other JPG files from the same folder generated "Invalid argument". Surprisingly, running it a 4th time with the first file now generated an "Invalid argument" exception as well! Yet when I use all these JPG files as arguments to the bundled exifprint.exe, all the JPG files display EXIF data.

I'm using the latest updates to Win 11 Pro and Visual Studio Community 2022 17.7. I'm sure my issue has something to do with a configuration error, but I've spent the last few hours trying to figure out what the problem is, obviously with no luck. I'm hoping someone out there can give me some insight into what is causing me so much grief.

Thanks!

**Comment by clanmills** [comment]
I'm as surprised as you to hear that there's a difference between the supplied exifprint.exe and your console app+samples/exifprint.cpp.

There is an option `exifprint.exe --version` to call `Exiv2::dumpLibraryInfo()`, which generates a report about the library's build.  Here's the output on macOS.

```bash
546 rmills@rmillsm1:~/gnu/github/exiv2/0.27-maintenance $ env DYLD_LIBRARY_PATH=build/lib build/bin/exifprint --version
exiv2=0.27.7
platform=apple
compiler=Clang
bits=64
dll=1
debug=0
cplusplus=199711
version=Apple LLVM 15.0.0 (clang-1500.0.40.1)
date=Sep 24 2023
time=11:36:24
processpath=/Users/rmills/gnu/github/exiv2/0.27-maintenance/build/bin
package_name=exiv2
curl=0
executable=/Users/rmills/gnu/github/exiv2/0.27-maintenance/build/bin/exifprint
library=/Users/rmills/gnu/github/exiv2/0.27-maintenance/build/lib/libexiv2.0.27.7.dylib
library=/usr/lib/system/libsystem_kernel.dylib
library=/usr/lib/system/libsystem_platform.dylib
library=/usr/lib/system/libsystem_pthread.dylib
have_inttypes=0
have_libintl=0
have_lensdata=1
...
```

I expect you'll find a significant difference between the supplied exifprint.exe and your console+exifprint.cpp.
